test/fakeintake/server/serverstore/db.go

```diff
@@ -46,16 +48,21 @@ type sqlMetrics struct {
 }
 
 // newSQLStore initializes a new payloads store with an SQLite DB
-func newSQLStore() *sqlStore {
-	p := os.Getenv("SQLITE_DB_PATH")
-	if p == "" {
-		f, err := os.CreateTemp("", defaultDBPath)
-		if err != nil {
-			log.Fatal(err)
-		}
-		p = f.Name()
+func newSQLStore(sqliteDbPath string) *sqlStore {
+	envOverrideDbPath := os.Getenv(SqliteDbPathEnv)
+
+	if envOverrideDbPath != "" {
+		sqliteDbPath = envOverrideDbPath
+	} else if sqliteDbPath == "" {
+		sqliteDbPath = defaultDBPath
+	}
```

Comment:
This could be futher improved with something like:

```golang
var path

if env != "" {
 path = env
} else if userPath != "" {
  path = userPath
} else {
  path = default
}
```

to make it clearer in the logical order of override.